src/js/abstract/base-component-global.js

```diff
@@ -23,15 +24,25 @@ export default class BaseComponentGlobal extends BaseComponent {
    * @return {type}        description
    */
   static appendGlobalStyles(styles, nodeName = BaseComponent.uuidv4()) {
-    if (styles) {
-      if (!memory[styles]) {
-        const styleNode = document.createElement('style');
-        const styleText = document.createTextNode(styles);
-        styleNode.appendChild(styleText);
-        styleNode.setAttribute('data-c-name', nodeName.toLowerCase());
-        document.querySelector('head').insertAdjacentElement('afterbegin', styleNode);
-        memory[styles] = true;
+    if (styles && !memory[styles]) {
+      let target = document.head;
+      const styleNode = document.createElement('style');
+      const styleText = document.createTextNode(styles);
+
+      styleNode.appendChild(styleText);
+      styleNode.setAttribute('data-c-name', nodeName.toLowerCase());
+
+      if (!styleTags) {
+        styleTags = document.getElementsByTagName('style');
```

Comment:
@LucaMele 
Since the result of `getElementsByTagName` is live, we can lazily query this just once.
Downside, this reference will remain forever until the browser window is closed...

https://developer.mozilla.org/en-US/docs/Web/API/Element/getElementsByTagName

Reply:
same is true for the `memory`, I guess it's easier to cache by it's `nodeName` only. Not the whole CSS

Reply:
i think so yes